src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/VilkårperiodeService.kt

```diff
@@ -47,6 +57,37 @@ class VilkårperiodeService(
         )
     }
 
+    fun hentVilkårperioderResponse(behandlingId: UUID): VilkårperioderResponse {
+        val grunnlagsdataVilkårsperioder = hentEllerOpprettGrunnlag(behandlingId)
+
+        return VilkårperioderResponse(
+            vilkårperioder = hentVilkårperioderDto(behandlingId),
+            grunnlag = grunnlagsdataVilkårsperioder,
+        )
+    }
+
+    private fun hentEllerOpprettGrunnlag(behandlingId: UUID): VilkårperioderGrunnlag {
+        val grunnlagsdataAktiviteter =
+            vilkårperioderGrunnlagRepository.findByBehandlingId(behandlingId) ?: opprettGrunnlagsdata(behandlingId)
```

Comment:
Burde opprettGrunnlagsdata gjøres når man oppretter behandlingen eller kun hvis behandlingen har riktig status? 
Hvis man går in på eldre behandlingen når så vil man opprette grunnlaget når man går inn på inngangsvilkåren

Reply:
Godt innspill! Fikser